chapters/batching.rst

```diff
@@ -0,0 +1,131 @@
+
+In the case where this decorator is used, the method needs to be implemented considering a single input.
+
+The single_multi_input is the equivalent decorator required when multiple input arguments are required by the :term:`Executor` method.
+
+.. highlight:: python
+.. code-block:: python
+
+    from jina.executors.crafters import BaseCrafter
+    from jina.executors.decorators import single
+    class CapitalizerCrafter(BaseCrafter):
+
+        @single
+        def craft(self, text: str, *args, **kwargs):
+            return {'text': text.capitalize()}
+
+    crafter = CapitalizerCrafter()
+
+    batch_of_docs = ['first', 'second', 'third']
+    crafted_results = crafter.craft(batch_of_docs)
+    assert crafted_results[0]['text'] == 'First'
+    assert crafted_results[1]['text'] == 'Second'
+    assert crafted_results[2]['text'] == 'Third'
+
+
+batching and batching_multi_input
+------------
+The batching decorator ensures that data is processed in batches of a given size. This is useful when an :term:`Executor` can benefit
+from processing multiple inputs at a time.
+
+In the case where this decorator is used, the method needs to be implemented considering input comes in batches.
+
+The batching_multi_input is the equivalent decorator required when multiple input arguments are required by the :term:`Executor` method.
+
+.. highlight:: python
+.. code-block:: python
+
+    from jina.executors.crafters import BaseCrafter
+    from jina.executors.decorators import batching
+
+    class CapitalizerCrafter(BaseCrafter):
+
+        @batching
+        def craft(self, text: Iterable[str], *args, **kwargs):
+            return [{'text': t.capitalize()} for t in text]
+
+    crafter = CapitalizerCrafter()
+
+    batch_of_docs = ['first', 'second', 'third']
+    crafted_results = crafter.craft(batch_of_docs)
+    assert crafted_results[0]['text'] == 'First'
+    assert crafted_results[1]['text'] == 'Second'
+    assert crafted_results[2]['text'] == 'Third'
+
+
+Batching vs Single
+------------
+The usage of `batching` or `single` does not affect the correct functionality, from the :term:`Driver` point of view it is the same, it can provide
+call them with a set of inputs, and gets a set of output in return. The only changes are encapsulated in the :term:`Executor` itself.
+
+Batch Size
+------------
+When using `batching` decorator, one may wonder how an :term:`Executor` can control the batch_size. There are 3 ways an Executor can define this.
+
+- Provide it directly in the batching decorator as a hardcoded value:
+    .. highlight:: python
+    .. code-block:: python
+
+        class CapitalizerCrafter(BaseCrafter):
+
+            @batching(batch_size=64)
+            def craft(self, text: Iterable[str], *args, **kwargs):
+                pass
+
+- Make it an attribute of the :term:`Executor` explicitly:
+
```

Comment:
Need to remove this blank line to make it consistent with the other 2 ways

Reply:
Do what u need to do